### PR TITLE
chore: Remove uses of deprecated fields in docs and manifests

### PR DESCRIPTION
**Pull request by terrytangyuan**
Signed-off-by: terrytangyuan <terrytangyuan@gmail.com>

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).

### CODE REVIEW

**Review by alexec — COMMENTED** [review]
I think we might have missed the v3.0 boat for this. I suggest we don't do it.

**Comment by terrytangyuan** [comment]
> I think we might have missed the v3.0 boat for this. I suggest we don't do it.

Agreed that it may not be good timing. I've updated the PR to just remove uses of deprecated fields in docs and manifests instead.

File: examples/gc-ttl.yaml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 # This example shows the ability to automatically delete workflows after a specified time period
 # after the workflow completes. The TTLStrategy sets the strategy for how long workflows that are
-# successful, not successful, or completed should live. The former ttlSecondsAfterFinished will be
-# deprecated  and will be replaced with TTLStrategy.SecondsAfterCompleted.
```
Comment (terrytangyuan) [inline examples/gc-ttl.yaml:4]: This field is already removed.

File: manifests/quick-start-minimal.yaml [hunk 2]
```diff
@@ -647,7 +647,6 @@ data:
       scope: sensor-logs
       url: http://logging-facility?namespace=${metadata.namespace}&podName=${metadata.name}&startedAt=${status.startedAt}&finishedAt=${status.finishedAt}
   metricsConfig: |
-    disableLegacy: true
```
Comment (terrytangyuan) [inline manifests/quick-start-minimal.yaml:650]: This field is not used and will be have no effect.

**Review by alexec — APPROVED** [review]

**Pull request merged**